### PR TITLE
ENH: add example for resampling transformed images to be axis aligned

### PR DESCRIPTION
**Pull request by dzenanz**

### CODE REVIEW

**Review by thewtex — CHANGES_REQUESTED** [review]
We do not want to use non-orthogonal bases in the Direction matrix.

Instead this should

- Use ImageSpatialObject to define the sampling grid for the output. Use the Transform to move the ImageSpatialObject. 
- Be written in Python

File: Data/axisAlignedResample.cmd [hunk 1]
```diff
@@ -0,0 +1,28 @@
+C:\Dev\HASI-ex\RelWithDebInfo\resampleAxisAligned.exe C:\Dev\HASI\Data\901-L.nrrd       C:\Dev\HASI\Tibias\901-L-label.nrrd       C:\Dev\HASI\Tibias\901-L-landmarks.tfm       C:\Dev\HASI\Tibias\901-L-AA.nrrd       C:\Dev\HASI\Tibias\901-L-AA-label.nrrd
+C:\Dev\HASI-ex\RelWithDebInfo\resampleAxisAligned.exe C:\Dev\HASI\Data\901-R.nrrd       C:\Dev\HASI\Tibias\901-R-label.nrrd       C:\Dev\HASI\Tibias\901-R-landmarks.tfm       C:\Dev\HASI\Tibias\901-R-AA.nrrd       C:\Dev\HASI\Tibias\901-R-AA-label.nrrd
+C:\Dev\HASI-ex\RelWithDebInfo\resampleAxisAligned.exe C:\Dev\HASI\Data\902-L.nrrd       C:\Dev\HASI\Tibias\902-L-label.nrrd       C:\Dev\HASI\Tibias\902-L-landmarks.tfm       C:\Dev\HASI\Tibias\902-L-AA.nrrd       C:\Dev\HASI\Tibias\902-L-AA-label.nrrd
+C:\Dev\HASI-ex\RelWithDebInfo\resampleAxisAligned.exe C:\Dev\HASI\Data\902-R.nrrd       C:\Dev\HASI\Tibias\902-R-label.nrrd       C:\Dev\HASI\Tibias\902-R-landmarks.tfm       C:\Dev\HASI\Tibias\902-R-AA.nrrd       C:\Dev\HASI\Tibias\902-R-AA-label.nrrd
```
Comment (thewtex) [inline Data/axisAlignedResample.cmd:4]: These should be removed, all processing should be Python scripts calling Python functions in the `hasi` package.

**Review by thewtex — CHANGES_REQUESTED** [review]

File: examples/ResampleAxisAligned.ipynb [hunk 2]
```diff
@@ -0,0 +1,166 @@
+{
```
Comment (thewtex) [inline examples/ResampleAxisAligned.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.        # direct_transform.ApplyToImageMetadata(labels)    
</pre><p>Do these require itk-5.3? If so, could a comment please be added accordingly?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/KitwareMedical/HASI/pull/38/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (dzenanz) [reply examples/ResampleAxisAligned.ipynb:1]: I added a comment there and force-pushed.

**Review by thewtex — APPROVED** [review]
:1234: :snake: :+1:

**Pull request merged**